ocp_resources/datavolume.py

```diff
@@ -106,40 +107,47 @@ def __init__(
         self.secret = secret
         self.content_type = content_type
         self.size = size
-        self.access_modes = access_modes
+        self.access_modes = (
+            access_modes if use_storage_api or access_modes else self.AccessMode.ROX
+        )
         self.storage_class = storage_class
-        self.volume_mode = volume_mode
+        self.volume_mode = (
+            volume_mode if use_storage_api or volume_mode else self.VolumeMode.FILE
+        )
         self.hostpath_node = hostpath_node
         self.source_pvc = source_pvc
         self.source_namespace = source_namespace
         self.multus_annotation = multus_annotation
         self.bind_immediate_annotation = bind_immediate_annotation
         self.preallocation = preallocation
+        self.use_storage_api = use_storage_api
 
     def to_dict(self):
         res = super().to_dict()
+        api_to_use = "storage" if self.use_storage_api else "pvc"
```

Comment:
Should be set after line 129, no need to save something that may never be used before needed.

Reply:
Ack